go/arrow/array/fixed_size_list.go

```diff
@@ -207,6 +207,10 @@ func (b *FixedSizeListBuilder) Append(v bool) {
 func (b *FixedSizeListBuilder) AppendNull() {
 	b.Reserve(1)
 	b.unsafeAppendBoolToBitmap(false)
+	// require to append this due to value indexes
+	for i := int32(0); i < b.n; i++ {
+		b.values.AppendNull()
+	}
```

Comment:
I think it might be extracted into a separate PR.
But, it'll have to be merged prior to this one then, as the quirky code was found during the testing.

Reply:
rebased